opencog/nlp/learn/scm/link-pipeline.scm

```diff
@@ -461,36 +464,19 @@
  it parsed, and then updates the counts for the observed words and word
  pairs.
 "
-	; try-catch wrapper around the counters. Due to a buggy RelEx
-	; (see documentation for `word-inst-get-word`), the function
-	; `update-clique-pair-counts` might throw.  If it does throw,
-	; then avoid doing any counting at all for this sentence.
-	;
-	; Note: update-clique-pair-counts commented out. If you want this,
-	; then uncomment it, and adjust the length.
-	; Note: update-disjunct-counts commented out. It generates some
-	; data, but none of it will be interesting to most people.
-	(define (update-counts sent)
-		(catch 'wrong-type-arg
```

Comment:
Have you actually tested, to make sure that there are no other code-paths that might result in a throw?  Large parts of opencog use exceptions in lots and lots of places, for performance:  It is much much cheaper to catch an exception every now and then, than it is to make millions of if-statement checks of return values. That is, relex is not the only thing that throws, there are other blocks of code that also use exceptions to report errors.

Reply:
I have parsed a few corpora without this exception, and I haven't found a throw here. Any other particular test you recommend?
The arguments passed to the functions that go inside the exception are a linkage from LG "any" (no throws detected in my experiments), and parameters for the whole run; if the latest are the wrong type, I don't want to skip one sentence, but getting a throw to correct my parameter.
Also, there's another catch one function higher in local-process, so I feel it's safe to remove this. Am I right?

Reply:
OK